#163620127 GET /parties/:id

frankly034 opened the pull request:
#### What does this PR do?
Have api endpoint for GET /parties/:id

#### Description of Task to be completed?
* Create party model function to fetches a specific party from the resource
* Create party controller function to manage fetch specific party requests
* Create party route to handle GET /parties/:id request
* Create parties/:id test
- ensure non-numeric id return status: 400
- ensure empty search to return status: 404
- return 200 and data if found

#### How should this be manually tested?
After cloning the repo, run 'npm run test'

#### Any background context you want to provide?
* Http test were done using supertest and chai expect library

#### What are the relevant pivotal tracker stories?
#163620127

frankly034 committed:
chore(model) add getAParty()

- return a party from the dummy data of parties with specific id
[Delivers #163620127]

frankly034 committed:
chore(middleware): add getAParty()

- ensures that a number parameter is passed in the request
- returns 400 if id isNaN
[Delivers #163620127]

frankly034 committed:
chore(controller) add getParty()

- ensure that a party was found else return 404
- return found party
[Delivers #163620127]

frankly034 committed:
bug(typo): Wrong spelling getAParty

[Delivers #163620127]

frankly034 committed:
bug(id error): Recurring dummy data id

- ensure unique dummy data id
[Delivers #163620127]

frankly034 committed:
feat(route): add GET /parties/:id

- makes the route available for comsumption
[Delivers #163620127]

frankly034 committed:
feat(test): add test for GET /parties/:id

- ensure that non-numeric req id returns a 400
- ensure that an empty result returns 404
- return party if found
[Delivers #163620127]

frankly034 committed:
bug(PascalCase): refactor file names

- refactor class file names to PascalCase
[Delivers #163620127]

frankly034 commented:
@Lumexralph guess your having a good day. Here's another PR

Lumexralph commented on server/middlewares/partyMiddleware.js:
```diff
@@ -7,6 +7,15 @@ class PartyMiddleware {
+
+  static getAParty(req, res, next) {
+    const { id } = req.params;
+    if (isNaN(parseInt(id, 10))) {
```
@frankly034 Good work so far! please use the `Number.isNaN` as against the `isNaN` global method you used

frankly034 replied:
Thank you very much. Was thinking what that eslint error meant.

frankly034 replied:
Resolved some conflicts.

frankly034 committed:
bug(global method): replace isNaN

- replace isNaN with Number.isNaN
- because the former is a global method
[Delivers #163620127]

frankly034 committed:
Merge branch 'develop' into ft-get-specific-party-endpoint-163620127

Lumexralph reviewed (approved):
Good Job! This work looks good to me.
